SalesforceSDK/Analytics/Store/EventStoreManager.cs

```diff
@@ -0,0 +1,224 @@
+﻿/*
+ * Copyright (c) 2016, salesforce.com, inc.
+ * All rights reserved.
+ * Redistribution and use of this software in source and binary forms, with or
+ * without modification, are permitted provided that the following conditions
+ * are met:
+ * - Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * - Neither the name of salesforce.com, inc. nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using Newtonsoft.Json.Linq;
+using Salesforce.SDK.Analytics.Model;
+using Salesforce.SDK.Core;
+using Salesforce.SDK.Logging;
+using PCLStorage;
+
+namespace Salesforce.SDK.Analytics.Store
+{
+    public class EventStoreManager : IEventStoreManager
+    {
+        private string _filenameSuffix;
+        private readonly IFolder _rootDir;
+        private string _encryptionKey;
+        private bool _isLoggingEnabled = true;
+        private int _maxEvents = 1000;
+        private static ILoggingService LoggingService => SDKServiceLocator.Get<ILoggingService>();
+        public string FilenameSuffix
+        {
+            get { return _filenameSuffix; }
```

Comment:
why have an explicit private backer field on these?  Could just do `public string FilenameSuffix{get;set;}` and that will have an implicit private backer field.  Usually you set it explicitly if you are going to raise a property changed event or something.